### PR TITLE
feature/autogenerate_mongo_db_ids-GDEV-160

### PR DESCRIPTION
**Pull request by alamgirmani**

### CODE REVIEW

**Review by ac-jorellanaf — APPROVED** [review]

File: sandbox_request/dao/request.py [hunk 1]
```diff
@@ -32,57 +33,75 @@ async def get_all_requests() -> List[Request]:
...
+async def get_next_request_id(counter, collection_name):
+    """
+    This method generates the sequence id for the MongoDB document
+    """
+    db_connect = DBConnect()
+    collection = await db_connect.get_collection(name=counter)
+    document = await collection.find_one({"_id": collection_name})  # type: ignore
+    collection.update_one({"_id": collection_name}, {"$inc": {"value": 1}})  # type: ignore
+    await db_connect.close_db()
+    return "REQ:" + str(document["value"] + 1)
```
Comment (deepakunni3) [inline sandbox_request/dao/request.py:80]: I would recommend padding the string with zeroes to resemble our ID formatting.
```py
return f"REQ:{document['value']:07}"
```

**Review by deepakunni3 — CHANGES_REQUESTED** [review]
@alamgirmani Looks good. Just a comment above regarding the ID formatting.

File: examples/requests.json [hunk 2]
```diff
@@ -2,29 +2,29 @@
   "requests": [
     {
       "dataset_id": "DAT:0000001",
-      "id": "REQ:0000001",
       "purpose": "Science",
+      "request_id": "REQ:0000001",
```
Comment (KerstenBreuer) [inline examples/requests.json:6]: Why have you renamed the "id" to "request_id"? I think it's a common standard to name the id of an collection object as "id" and any references to foreign collections "{collection_name}_id".

No matter, what we decide on, I think, we should be consistent across the request and metadata service.

File: sandbox_request/dao/request.py [hunk 3]
```diff
@@ -32,57 +33,75 @@ async def get_all_requests() -> List[Request]:
...
+    data.request_id = request_id
     await collection.insert_one(data.dict())  # type: ignore
     request = await get_request(request_id)
-    send_mail("data_steward", "request_made")
+    send_mail(data.user_id, "requested")
+    await db_connect.close_db()
     return request
 
 
+async def get_next_request_id(counter, collection_name):
```
Comment (KerstenBreuer) [inline sandbox_request/dao/request.py:71]: Could we move this function to the top, as this is a dependency of `add_request`?

**Review by KerstenBreuer — CHANGES_REQUESTED** [review]
Looks good.

File: sandbox_request/dao/request.py (hunk 3 above)
Comment (alamgirmani) [reply sandbox_request/dao/request.py:71]: Not sure if the positioning of the methods matters, is this a standard convention?

File: examples/requests.json (hunk 2 above)
Comment (alamgirmani) [reply examples/requests.json:6]: It was originally named "id" and we can revert back to "id". We will still have a mongo generated "_id". I will make this change.

File: sandbox_request/dao/request.py (hunk 1 above)
Comment (alamgirmani) [reply sandbox_request/dao/request.py:80]: Thank you, I will reformat it as recommended

Comment (alamgirmani) [reply sandbox_request/dao/request.py]: pushed the changes

File: examples/requests.json (hunk 2 above)
Comment (alamgirmani) [reply examples/requests.json]: pushed the changes

**Review by KerstenBreuer — APPROVED** [review]

**Pull request merged**